packages/core/strapi/lib/types/core-api/controller.d.ts

```diff
@@ -0,0 +1,52 @@
+import type { Common, CoreApi, Utils } from '@strapi/strapi';
+import type { ExtendableContext } from 'koa';
+
+/**
+ * Base Core-API controller type
+ *
+ * TODO: Make use of the T generic to type the other methods based on the given content type
+ */
+export interface Base {
+  // TODO: Use actual entities instead of regular object
+  transformResponse<U, P>(data: U, meta: object): P;
+  sanitizeOutput<U>(data: U, ctx: ExtendableContext): Promise<U>;
+  sanitizeInput<U>(data: U, ctx: ExtendableContext): Promise<U>;
+  sanitizeQuery<U>(data: U, ctx: ExtendableContext): Promise<U>;
```

Comment:
```suggestion
  transformResponse<TData, TResponse>(data: TData, meta: object): TResponse;
  sanitizeOutput<TData>(data: TData, ctx: ExtendableContext): Promise<TData>;
  sanitizeInput<TData>(data: TData, ctx: ExtendableContext): Promise<TData>;
  sanitizeQuery<TData>(data: TData, ctx: ExtendableContext): Promise<TData>;
```
Oops, missed this one before